Update: AdvancePaymentController Improvements

EQC16 opened the pull request:
Description

This update enhances the management, filtering, and reporting of advance payments in the system. The data retrieval logic has been optimized, the code has been refactored for better readability, and new validations and logging have been added. The visual presentation in the admin panel and PDF reports has also been improved.

Main Changes
•	🔧 Full controller refactor:
o	Improved payment retrieval using is_future_payment = 1.
o	Reorganized private methods with modern, cleaner syntax.
o	Enhanced getChartData() to dynamically generate charts for the last three months.
o	Added more detailed logging for better traceability.
•	📈 Improved PDF reports:
o	Adjusted chart layouts (advancePaymentGraphReport.blade.php) for better visual balance.
o	Added total amounts, clearer labels, and default date handling.
•	🌱 Seeder improvements (AdvancePaymentsSeeder):
o	Added is_future_payment field to identify advance payments.
o	Generated more realistic created_at and updated_at timestamps.
•	🧭 Interface updates (adminlte.php and views):
o	Changed menu label from “Gráficos de Pagos Adelantados” to “Panel de Pagos Adelantados”.
o	Adjusted chart spacing in index.blade.php for a cleaner display.

Impact

These changes provide more accurate control over advance payments, improve historical data visualization, and deliver more informative and well-structured reports — enhancing decision-making within the administrative panel.

EQC16 committed:
The advance payment file has been updated and corrected in PDF format for better visibility, and the Seeder has been modified for better management in the database and its use in graphics with the controller.

And-Y21 commented on app/Http/Controllers/AdvancePaymentController.php:
```diff
@@ -2,22 +2,21 @@
 use Illuminate\Support\Facades\DB;
-use Carbon\Carbon;
-use Str;
-use App\Models\Debt;
```
Todo ese movimiento no es necesario. Quitas el modelo de deudas y pagos y lo vueleves a poner abajo.

And-Y21 commented on app/Http/Controllers/AdvancePaymentController.php:
```diff
@@ -2,22 +2,21 @@
-{
-    public function index(Request $request)
-    {
+class AdvancePaymentController extends Controller {
```
La estructura de las llaves ya se han visto en otros controladores
funcion
{ 
    // codigo
}

EQC16 replied:
vale ya lo estructure como pidio

And-Y21 commented on app/Http/Controllers/AdvancePaymentController.php:
```diff
@@ -27,117 +26,114 @@ public function index(Request $request)
 
-    public function destroy($id)
-    {
+    public function destroy($id) {
```
Lo mismo para estas funciones

EQC16 replied:
ya lo acomode como deveria de ir

EQC16 committed:
Corrected indentation in the AdvancePaymentController.php file

The file indentation is updated for better code readability and unnecessary log messages are removed.

EQC16 commented on app/Http/Controllers/AdvancePaymentController.php:
```diff
@@ -2,22 +2,21 @@
-use Str;
-use App\Models\Debt;
 use Illuminate\Support\Facades\Log;
-use App\Models\WaterConnection;
```
Listo ya esta correjido
